static/help-docs/users.md

```diff
@@ -79,3 +79,125 @@ called "Kleisli" in some contexts. This may also change in the future; see
 [`sequence`]: https://pursuit.purescript.org/packages/purescript-foldable-traversable/docs/Data.Traversable#t:Traversable
 [Modules page in the documentation repository]: https://github.com/purescript/documentation/blob/master/language/Modules.md
 [`Star`]: https://pursuit.purescript.org/packages/purescript-profunctor/docs/Data.Profunctor.Star#t:Star
+
+## Kind Signatures
+
+### Explicit and Inferred
+
+Data, newtype, type synonym, and type class declarations all have kind
+signatures. These signatures are either explicit (i.e. developer-defined)
+or implicit (i.e. compiler-inferred). For example
+
+```purescript
+-- Explicit kind signature
+
+data ExplicitFoo :: forall k. k -> Type
+data ExplicitFoo a = ExplicitFoo
+
+-- Implicit kind signature
+                                                                  {-
+data ImplicitFoo :: forall k. k -> (Type -> Type) -> Type         -}
+data ImplicitFoo a f = ImplicitFoo (f Int)
```

Comment:
Perhaps we could do something to make it a little clearer the signature is being inferred by something like:

```purs
{- The compiler will infer this kind signature for the below type:
data ImplicitFoo :: forall k. k -> (Type -> Type) -> Type
-}
data ImplicitFoo a f = ImplicitFoo (f Int)

Reply:
@kl0tl Here's what I was doing above originally. Thomas is the one who suggested the edit that you commented on below.

Which is preferred?